packages/turbo-kovan-subgraph/src/helpers/CommonHandlers.ts

```diff
@@ -0,0 +1,129 @@
+  event: LiquidityChanged,
+  positionFromAddLiquidity: boolean,
+  sharesReturned: BigInt,
+  outcomeId: BigInt
+): void {
+  let id = event.transaction.hash.toHexString() + "-" + outcomeId.toHexString();
+  let positionBalanceEntity = getOrCreatePositionBalance(id, true, false);
+  let senderId = event.params.user.toHexString();
+  getOrCreateSender(senderId);
+
+  positionBalanceEntity.positionFromAddLiquidity = positionFromAddLiquidity;
+  positionBalanceEntity.positionFromRemoveLiquidity = !positionFromAddLiquidity;
+  positionBalanceEntity.hasClaimed = false;
+  positionBalanceEntity.transactionHash = event.transaction.hash.toHexString();
+  positionBalanceEntity.timestamp = event.block.timestamp;
+  positionBalanceEntity.outcomeId = bigIntToHexString(outcomeId);
+  positionBalanceEntity.sender = senderId;
+
+  let collateralBigDecimal = event.params.collateral.toBigDecimal().div(USDC_DECIMALS);
+  let sharesReturnedBigDecimal = (sharesReturned.toBigDecimal()).div(SHARES_DECIMALS);
+  positionBalanceEntity.shares = bigIntToHexString(sharesReturned);
+  positionBalanceEntity.sharesBigInt = sharesReturned;
+  positionBalanceEntity.sharesBigDecimal = sharesReturnedBigDecimal;
+  positionBalanceEntity.initCostUsd = bigIntToHexString(event.params.collateral);
+  positionBalanceEntity.initCostUsdBigInt = event.params.collateral;
+  positionBalanceEntity.initCostUsdBigDecimal = collateralBigDecimal;
+  positionBalanceEntity.avgPrice = collateralBigDecimal.div(sharesReturnedBigDecimal);
+
+  positionBalanceEntity.save();
+}
+
+export function handlePositionFromClaimWinningsEvent(
+  event: WinningsClaimed,
+): void {
+  let id = event.transaction.hash.toHexString() + "-" + event.params.winningOutcome.toHexString();
+  let positionBalanceEntity = getOrCreatePositionBalance(id, true, false);
+  let senderId = event.params.receiver.toHexString();
+  getOrCreateSender(senderId);
+
+  positionBalanceEntity.positionFromAddLiquidity = false;
+  positionBalanceEntity.positionFromRemoveLiquidity = false;
+  positionBalanceEntity.hasClaimed = true;
+  positionBalanceEntity.transactionHash = event.transaction.hash.toHexString();
+  positionBalanceEntity.timestamp = event.block.timestamp;
+  positionBalanceEntity.outcomeId = event.params.winningOutcome.toHexString();
+  positionBalanceEntity.sender = senderId;
+
+  let payoutBigDecimal = event.params.payout.toBigDecimal().div(USDC_DECIMALS);
+  let amountBigDecimal = event.params.amount.toBigDecimal().div(SHARES_DECIMALS);
+  positionBalanceEntity.shares = bigIntToHexString(event.params.amount);
+  positionBalanceEntity.sharesBigInt = event.params.amount;
+  positionBalanceEntity.sharesBigDecimal = amountBigDecimal;
+  positionBalanceEntity.initCostUsd = bigIntToHexString(event.params.payout);
+  positionBalanceEntity.initCostUsdBigInt = event.params.payout;
+  positionBalanceEntity.initCostUsdBigDecimal = payoutBigDecimal;
+  positionBalanceEntity.avgPrice = payoutBigDecimal.div(amountBigDecimal);
```

Comment:
When winningsClaimed log comes in don't change existing initCostUsdBigInt, initCostUsdBigDecimal, avgPrice
They will be needed as is.

add properties 
`winningShares`
`payout`
`settlementFee`